rollbar-java/src/integTest/java/com/rollbar/notifier/RollbarITest.java

```diff
@@ -244,10 +268,163 @@ public void shouldNotifyAccessTokenRequiredError() {
 
     rollbar.error(ERROR_MESSAGE);
+
+    listener.assertCalled();
+  }
+
+  @Test
+  public void shouldNotifyInvalidResponseError() {
+    Sender sender = buildSender(getUrl(), null, null);
+
+    Config config = configBuilder
+        .accessToken(null)
+        .sender(sender)
+        .build();
+
+    stubFor(post(urlEqualTo("/api/1/item/"))
+        .willReturn(aResponse()
+            .withFault(Fault.MALFORMED_RESPONSE_CHUNK)));
+
+    SenderAssertions.ErrorAssertion
+        listener = SenderAssertions.assertSenderError(IOException.class);
+    sender.addListener(listener);
+
+    Rollbar rollbar = new Rollbar(config);
+
+    rollbar.error(ERROR_MESSAGE);
+
+    listener.assertCalled();
+  }
+
+  protected Sender buildSender(String url, String accessToken, Proxy proxy) {
+    SyncSender.Builder builder = new SyncSender.Builder()
+        .url(url);
+
+    if (accessToken != null) {
+      builder = builder.accessToken(accessToken);
+    }
+
+    if (proxy != null) {
+      builder = builder.proxy(proxy);
+    }
+
+    return builder.build();
+  }
+
+  private int getPort() {
+    return wireMockRule.port();
+  }
+
+  private String getUrl() {
+    return format(Locale.US, "http://localhost:%d/api/1/item/", getPort());
+  }
+
+  public static class BlockingListener implements SenderListener {
+    public AtomicBoolean done = new AtomicBoolean(false);
+    private final long timeout;
+
+    public BlockingListener(long timeoutMs) {
+      this.timeout = timeoutMs;
+    }
+
+    @Override
+    public void onResponse(Payload payload, Response response) {
+      done.set(true);
+    }
+
+    @Override
+    public void onError(Payload payload, Exception error) {
+      done.set(true);
+    }
+
+    public void clear() {
+      done.set(false);
+    }
+
+    public void block() throws InterruptedException, TimeoutException {
+      long remaining = timeout;
+      int sleepTime = 20;
+
+      while (!done.get()) {
+        if (remaining < 0) {
+          throw new TimeoutException("Timed out waiting for result");
+        }
+
+        Thread.sleep(sleepTime);
+        remaining -= sleepTime;
+      }
+    }
+  }
+
+  private ProxyServer buildProxyServer() {
+    // WireMock only supports 'CONNECT' for SSL requests, and injects some SSL data into the stream
+    // automatically after a request, which fails in our case since we're not actually using SSL.
+    // Netty's HTTP client always uses 'CONNECT' for proxying, even for HTTP, so we can't use
+    // WireMock as the proxy in the Reactor tests.
+    // See com.github.tomakehurst.wiremock.jetty94.ManInTheMiddleSslConnectHandler
+
+    HandlerCollection handlers = new HandlerCollection();
+    RequestCounter requestCounter = new RequestCounter();
+    handlers.addHandler(requestCounter);
+
+    ConnectHandler proxy = new ConnectHandler();
+
+    ServletContextHandler context = new ServletContextHandler(proxy, "/",
+        ServletContextHandler.SESSIONS);
```

Comment:
That's pretty standard java formatting for long lines, what do you think are the benefits of the updated version?

Reply:
That's clearly an advantage if one is used to that format, but it seems a bit marginal. And on the drawbacks there's  the additional vertical space it uses, and as I mentioned is not the most common formatting for java code. So it has pros and cons, and maybe we're getting too close to personal preference rather than objective benefits.

Reply:
I fully agree. It was just a suggestion, nothing compulsory to follow. The key is to be consistent and keep the same formatting along the project. And I'm fine with whatever style that you feel it suits here best.